### ISSUE TITLE
Flaky-test: ReplicatorTest.testReplicatorWithFailedAck

### ISSUE THREAD

**Issue by codelipenghui** [apache/pulsar#16500]
[example failure](https://github.com/apache/pulsar/runs/7263941429?check_suite_focus=true)

```
 Error:  Tests run: 52, Failures: 10, Errors: 0, Skipped: 21, Time elapsed: 375.654 s <<< FAILURE! - in org.apache.pulsar.broker.service.ReplicatorTest
  Error:  testReplicatorWithFailedAck(org.apache.pulsar.broker.service.ReplicatorTest)  Time elapsed: 32.549 s  <<< FAILURE!
  org.awaitility.core.ConditionTimeoutException: Condition with org.apache.pulsar.broker.service.ReplicatorTest was not fulfilled within 30 seconds.
  	at org.awaitility.core.ConditionAwaiter.await(ConditionAwaiter.java:167)
  	at org.awaitility.core.CallableCondition.await(CallableCondition.java:78)
  	at org.awaitility.core.CallableCondition.await(CallableCondition.java:26)
  	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:985)
  	at org.awaitility.core.ConditionFactory.until(ConditionFactory.java:954)
  	at org.apache.pulsar.broker.service.ReplicatorTest.testReplicatorWithFailedAck(ReplicatorTest.java:1418)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
  	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
  	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
  	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
  	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:132)
  	at org.testng.internal.InvokeMethodRunnable.runOne(InvokeMethodRunnable.java:45)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:73)
  	at org.testng.internal.InvokeMethodRunnable.call(InvokeMethodRunnable.java:11)
  	at java.base/java.util.concurrent.FutureTask.run(FutureTask.java:264)
  	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
  	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
  	at java.base/java.lang.Thread.run(Thread.java:833)
```

<!-- optionally provide the full stacktrace ->
<details>
<summary>Full exception stacktrace</summary>
<code><pre>
full exception stacktrace here
</pre></code>
</details>

**Comment by shibd** [comment]
+1 https://github.com/apache/pulsar/runs/7432985874?check_suite_focus=true

**Comment by nodece** [comment]
+1 https://github.com/apache/pulsar/runs/7526216129?check_suite_focus=true#step:10:1269

**Comment by codelipenghui** [comment]
The issue should be fixed by #17049
